docs/GraFx-Developers/connectors/media-connector/add-environment-options-to-your-connector/index.md

```diff
@@ -0,0 +1,88 @@
+# Add Environment Options to Your Connector
+
+This guide is about adding options that are set at the environment level for your Connector code. This is typically used of defining a value that will be shared among your Connector that is tied to every variable and document. For example, defining a base URL.
+
+## Requirements
+
+- Node.js or Bun.js installed
+- [Connector CLI](link-to-connector-cli-docs) tool
+- Environment Admin user with a Template Designer license
+- Completed [Build a Simple Media Connector](link-to-simple-media-connector-guide) tutorial or started from this [git project](https://github.com/seancrowe/simple-media-connector/tree/Build-a-Simple-Media-Connector)
+
+
+## Adding an Environment Option
+
+We'll add an option to define the base URL for the picsum.photos API that our Connector will use throughout the published environment.
+
+Modify the `options` property in your `package.json` to add the new option:
+
+```json
+{
+  ...
+  "config": {
+    "type": "media",
+    "options": {"baseURL": null},
+    "mappings": {}
+  },
+}
+```
+
+Setting the value to `null` requires you to define the value during publishing using the [Connector CLI](link-to-connector-cli-docs) tool. Alternatively, you can provide a default value instead of `null`. Note that only strings and booleans are allowed; numbers will be converted to strings.
+
+!!! Bug "Default Values Issue"
+    There is a known issue in the Connector CLI tool where default values are not applied correctly.
+
+## Implementing the Option in Code
+
+### Updating the `query` Method
+
+In the `query` method, access the new option via the `runtime` object. Update the fetch request by replacing the hardcoded base URL with the new option:
+
+```typescript
+const resp = await this.runtime.fetch(`https://${this.runtime.options["baseURL"]}/v2/list?page=1&limit=${limit}`, {
+  method: "GET"
+});
+```
+
+### Updating the `download` Method
+
+Similarly, update the `download` method:
+
+```typescript
+async download(
+  id: string,
+  previewType: Media.DownloadType,
+  intent: Media.DownloadIntent,
+  context: Connector.Dictionary
+): Promise<Connector.ArrayBufferPointer> {
+  switch (previewType) {
+    case "thumbnail": {
+      const picture = await this.runtime.fetch(`https://${this.runtime.options["baseURL"]}/id/${id}/${context.wide ? "400/" : ""}200`, { method: "GET" });
```

Comment:
same as above, this has the option for wide images which isn't set up in the base repo